graylog2-server/src/main/java/org/graylog/plugins/views/search/engine/suggestions/FieldValueSuggestionMode.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.views.search.engine.suggestions;
+
+public enum FieldValueSuggestionMode {
+    ON,
+    TEXTUAL_ONLY,
+    OFF;
+
+    public static FieldValueSuggestionMode fromString(final String stringValue) {
+        if (ON.toString().equalsIgnoreCase(stringValue)) {
+            return ON;
+        }
+        if (TEXTUAL_ONLY.toString().equalsIgnoreCase(stringValue)) {
+            return TEXTUAL_ONLY;
+        }
+        return OFF;
```

Comment:
I'd suggest throwing an exception here, instead of defaulting to OFF.  Especially when the default in Configuration.java is ON. The valueOf of an upper-cased stringValue would simplify the logic further.

Reply:
I will get rid of that method completely, your previous suggestion allows for that. Thanks!